### ISSUE TITLE
hangs at 100% cpu when holding the escape key too long

### ISSUE THREAD

**Issue by jackolantern** [neovim/neovim#13551]
- `nvim --version`: NVIM v0.4.4
- `vim -u DEFAULTS` (version: ) behaves differently? yes
- Operating system/version: arch
- Terminal name/version: kitty
- `$TERM`: xterm-kitty

### Steps to reproduce using `nvim -u NORC`

nvim -u NORC
Press and hold the escape key for approximately one minute.

### Actual behaviour

nvim will become unresponsive and the process needs to be kill -9ed.  On my laptop the process will use 100% cpu when it is in this state.

### Expected behaviour

for nothing to happen.

**Comment by vigoux** [comment]
In `nvim` master, holding the escape key for too long results in  :

```
nvim: ../src/nvim/tui/input.c :609 : handle_raw_buffer:  assertion « consumed <= input->read_stream.buffer->size » failed.
```

**Comment by teto** [comment]
@erw7 knows this part well

**Comment by erw7** [comment]
The problem with assertion failures in https://github.com/neovim/neovim/issues/13551#issuecomment-747720618  should be fixed in #13588.

**Comment by erw7** [comment]
@jackolantern If you try waiting for a few minutes without killing the process, does it become responsive? If so, it's not a bug, but your CPU is busy processing key events. In the meantime, why don't you review the key repeat settings?

**Comment by mortenbh** [comment]
On my machine I don't need to hold escape for a whole minute to get a crash; It's more like 5 seconds.

**Comment by justinmk** [comment]
libtermkey issue? https://github.com/neovim/neovim/pull/13588#issuecomment-750853018

> When `termkey_push_bytes()` returns -1, I thought it would be a good idea to call `termkey_getkey_force()` until there is enough space in the buffer, but I found a new bug in libtermkey.
> 
> ```vim
> nvim -u NORC
> :term nvim -u NORC
> :call chansend(b:terminal_job_id, ":let g:esc = 0 | nnoremap <silent><ESC> :let g:esc += 1<CR>\<CR>")
> :call chansend(b:terminal_job_id, repeat("\<ESC>", 50) .. ":echo g:esc\<CR>")
> 1 " This should be 50.
> ```
> 
> `termkey_getkey_force()` will only generate one `<ESC>` from `<ESC><ESC><ESC>...`. It seems that the only way to fix this problem completely is to fix libtermkey.
> 
> Since the actual input probably won't generate 255 `<ESC>` followed by `<F1>` and so on, we may as well merge this as a workaround until libtermkey is fixed.

**Comment by zeertzjq** [comment]
Fixed by #26309
